### PR TITLE
Fix E2E tests in WordPress 5.8

### LINKED ISSUE #8160: Fix E2E tests in WordPress 5.8
## Feature Description

The following E2E tests are failing in WordPress 5.8. 

```
● Web Stories Block › should insert a new web stories block

    TimeoutError: waiting for function failed: timeout 3000ms exceeded

      28 |  */
      29 | async function waitForInserterCloseAndContentFocus() {
    > 30 |   await page.waitForFunction(() =>
         |              ^
      31 |     document.body
      32 |       .querySelector('.block-editor-block-list__layout')
      33 |       ?.contains(document.activeElement)

      at new WaitTask (../../../node_modules/puppeteer/src/common/DOMWorld.ts:777:28)
      at DOMWorld.waitForFunction (../../../node_modules/puppeteer/src/common/DOMWorld.ts:699:22)
      at Frame.waitForFunction (../../../node_modules/puppeteer/src/common/FrameManager.ts:1241:28)
      at Page.waitForFunction (../../../node_modules/puppeteer/src/common/Page.ts:3152:29)
      at waitForInserterCloseAndContentFocus (../../e2e-test-utils/src/insertBlock.js:30:14)
      at insertBlock (../../e2e-test-utils/src/insertBlock.js:50:9)
      at Object.<anonymous> (specs/wordpress/webStoriesBlock.js:90:5)

  ● Web Stories Block › should insert a new web stories block and select story

    TimeoutError: waiting for function failed: timeout 3000ms exceeded

      28 |  */
      29 | async function waitForInserterCloseAndContentFocus() {
    > 30 |   await page.waitForFunction(() =>
         |              ^
      31 |     document.body
      32 |       .querySelector('.block-editor-block-list__layout')
      33 |       ?.contains(document.activeElement)
```

This is because the block insert has changed and inserter doesn't close as it once did. We will need to write some conditional logic here to get the `insertBlock` util to work in multiple versions of WP. 

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

TLDR, in WordPress 5.8+, the block needs to be focused, before this is out was automatically focused and the inserter closed. 

Much of these changes are copied from upstream.

### QA

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #8160

### CODE REVIEW

File: packages/e2e-test-utils/src/insertBlock.js [hunk 1]
```diff
@@ -45,7 +45,11 @@ async function waitForInserterCloseAndContentFocus() {
  */
 async function insertBlock(searchTerm) {
   await searchForBlock(searchTerm);
-  await expect(page).toClick('button span', { text: searchTerm });
+  const insertButton = await page.waitForXPath(
+    `//button//span[contains(text(), '${searchTerm}')]`
+  );
+  await insertButton.click();
```
Comment (swissspidy) [inline packages/e2e-test-utils/src/insertBlock.js]: Why does the `expect(page).toClick()` not work?

File: packages/e2e-test-utils/src/focusSelectedBlock.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+async function focusSelectedBlock() {
+  // Ideally there shouuld be a UI way to do this. (Focus the selected block)
+  await page.evaluate(() => {
+    wp.data
+      .dispatch('core/block-editor')
+      .selectBlock(
+        wp.data.select('core/block-editor').getSelectedBlockClientId(),
+        0
```
Comment (swissspidy) [inline packages/e2e-test-utils/src/focusSelectedBlock.js:24]: What's the `0` here? AFAIK `selectBlock` only takes one argument.

File: packages/e2e-test-utils/src/focusSelectedBlock.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+async function focusSelectedBlock() {
+  // Ideally there shouuld be a UI way to do this. (Focus the selected block)
```
Comment (swissspidy) [inline packages/e2e-test-utils/src/focusSelectedBlock.js:18]: Can't that be done by clicking on the block or something?

File: packages/e2e-test-utils/src/insertBlock.js [hunk 4]
```diff
@@ -45,7 +45,11 @@ async function waitForInserterCloseAndContentFocus() {
...
+  await focusSelectedBlock();
```
Comment (swissspidy) [inline packages/e2e-test-utils/src/insertBlock.js:46]: Instead of focusing the block, could we close the inserter manually somehow?

Comment (spacedmonkey) [reply packages/e2e-test-utils/src/insertBlock.js:46]: Copied from https://github.com/WordPress/gutenberg/blob/8499a38f6d5f0cdfd86c4bf17d42f37232c72125/packages/e2e-test-utils/src/inserter.js#L147-L162

File: packages/e2e-test-utils/src/focusSelectedBlock.js (hunk 2 above)
Comment (spacedmonkey) [reply packages/e2e-test-utils/src/focusSelectedBlock.js:24]: Copied from https://github.com/WordPress/gutenberg/blob/8499a38f6d5f0cdfd86c4bf17d42f37232c72125/packages/e2e-test-utils/src/inserter.js#L61-L76

File: packages/e2e-test-utils/src/insertBlock.js (hunk 4 above)
Comment (swissspidy) [reply packages/e2e-test-utils/src/insertBlock.js:46]: oh, in that case, since our versions of `insertBlock` and `waitForInserterCloseAndContentFocus` are now identical to the ones from Gutenberg, we can just use `insertBlock` from `@wordpress/e2e-test-utils` directly instead of copying everything.

Comment (spacedmonkey) [reply packages/e2e-test-utils/src/insertBlock.js:46]: Sadly we can't as the version of `@wordpress/e2e-test-utils` is a month out of date. Also 
> It works properly with the minimum version of Gutenberg 9.2.0 or the minimum version of WordPress 5.6.0.

We support, 5.5, so have maintain our own version of these functions. 

https://github.com/WordPress/gutenberg/tree/trunk/packages/e2e-test-utils

File: packages/e2e-test-utils/src/insertBlock.js [hunk 5]
```diff
@@ -46,6 +46,7 @@ async function waitForInserterCloseAndContentFocus() {
 async function insertBlock(searchTerm) {
```
Comment (swissspidy) [inline packages/e2e-test-utils/src/insertBlock.js:43]: Now need to remove the `Avoids using waitForInserterCloseAndContentFocus() [...]` comment above as it's no longer accurate.

File: packages/e2e-test-utils/src/insertBlock.js (hunk 4 above)
Comment (swissspidy) [reply packages/e2e-test-utils/src/insertBlock.js:46]: > Sadly we can't as the version of @wordpress/e2e-test-utils is a month out of date.

Is it? At first glance I don't see a difference from the package published on npm that we use and the Gutenberg `trunk` branch.

I don't mind maintaining our own version of certain functions if they differ in logic, but here they seem identical to me. But happy to be corrected.

Comment (spacedmonkey) [reply packages/e2e-test-utils/src/insertBlock.js:46]: This [packages](https://www.npmjs.com/package/@wordpress/e2e-test-utils) says, last published 1 month ago. 

Screenshot 2021-07-07 at 12 22 52

Inserter was updated 12 minutes ago. 

Screenshot 2021-07-07 at 12 22 45

What am I missing?

Comment (swissspidy) [reply packages/e2e-test-utils/src/insertBlock.js:46]: That seems fine to me? Yes there's a recently modified file in there, but I don't see how that would affect us. [That change](https://github.com/WordPress/gutenberg/pull/32935) is neither in core nor in the package version that we use. We also don't use any of the [currently unreleased](https://github.com/WordPress/gutenberg/blob/f3f532015d3200f8b5b84dc2978e423b72c1fa98/packages/e2e-test-utils/CHANGELOG.md#unreleased) new features.

----

Putting this another way: if we just remove our own version of `insertBlock` and use the version from `@wordpress/e2e-test-utils`, do the e2e tests still pass? If yes, I'd say we can safely remove our copy.

File: packages/e2e-test-utils/src/focusSelectedBlock.js (hunk 2 above)
Comment (swissspidy) [reply packages/e2e-test-utils/src/focusSelectedBlock.js:24]: Looks like `0` is the default argument here:

https://github.com/WordPress/gutenberg/blob/8499a38f6d5f0cdfd86c4bf17d42f37232c72125/packages/block-editor/src/store/actions.js#L205-L223

Could be safely removed

**Comment by spacedmonkey** [comment]
Closing in favour https://github.com/google/web-stories-wp/pull/8223
